primefaces/src/main/java/org/primefaces/component/dashboard/DashboardRenderer.java

```diff
@@ -50,27 +52,33 @@ public void encodeEnd(FacesContext context, UIComponent component) throws IOExce
         encodeScript(context, dashboard);
     }
 
-    protected void encodeMarkup(FacesContext contextr, Dashboard dashboard) throws IOException {
-        ResponseWriter writer = contextr.getResponseWriter();
-        String clientId = dashboard.getClientId(contextr);
+    protected void encodeMarkup(FacesContext context, Dashboard dashboard) throws IOException {
+        ResponseWriter writer = context.getResponseWriter();
+        String clientId = dashboard.getClientId(context);
+        boolean responsive = dashboard.isResponsive();
 
         writer.startElement("div", dashboard);
         writer.writeAttribute("id", clientId, "id");
-        String styleClass = dashboard.getStyleClass() != null ? Dashboard.CONTAINER_CLASS + " " + dashboard.getStyleClass() : Dashboard.CONTAINER_CLASS;
-        if (dashboard.isDisabled()) {
-            styleClass = styleClass + " ui-state-disabled";
-        }
+        String styleClass = getStyleClassBuilder(context)
+                .add(Dashboard.CONTAINER_CLASS)
+                .add(dashboard.getStyleClass())
+                .add(dashboard.isDisabled(), "ui-state-disabled")
+                .add(responsive, GridLayoutUtils.getFlexGridClass(true))
+                .build();
         writer.writeAttribute("class", styleClass, "styleClass");
         if (dashboard.getStyle() != null) {
             writer.writeAttribute("style", dashboard.getStyle(), "style");
         }
 
         DashboardModel model = dashboard.getModel();
         if (model != null) {
-            for (DashboardColumn column : model.getColumns()) {
+            for (DashboardWidget column : model.getWidgets()) {
```

Comment:
Thomas asked me to use index based for loops for performance reasons.

https://github.com/primefaces/primefaces/blob/master/CONTRIBUTING.md#performance-guidelines

Reply:
You are right but this was how the legacy code was for the last 10 years :).  I can fix it.